pkg/cluster/delete.go

```
@@ -377,6 +378,77 @@ func (m *manager) deleteClusterMsiCertificate(ctx context.Context) error {
+			return nil
+		}
+	}
+
+	platformWIRolesByRoleName := m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()
+	platformWorkloadIdentities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+
+	for _, identity := range platformWorkloadIdentities {
+		_, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
+			continue
+		}
+
+		identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+		if err != nil {
+			return err
+		}
+
+		platformWIRole, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
+			continue
+		}
+
+		for _, sa := range platformWIRole.ServiceAccounts {
+			federatedIdentityCredentialResourceName, err := m.getPlatformWorkloadIdentityFederatedCredName(sa, identity)
+			if err != nil {
+				m.log.Errorf("failed to get federated identity credential name for %s: %v", identity.ResourceID, err)
+				continue
+			}
+
+			_, err = m.clusterMsiFederatedIdentityCredentials.Get(
+				ctx,
+				identityResourceId.ResourceGroup,
+				identityResourceId.ResourceName,
+				federatedIdentityCredentialResourceName,
+				&armmsi.FederatedIdentityCredentialsClientGetOptions{},
+			)
+
+			// If the federated identity credentials do not exist, we can skip the deletion
+			if err != nil {
+				continue
+			}
+
+			_, err = m.clusterMsiFederatedIdentityCredentials.Delete(
+				ctx,
+				identityResourceId.ResourceGroup,
+				identityResourceId.ResourceName,
+				federatedIdentityCredentialResourceName,
+				&armmsi.FederatedIdentityCredentialsClientDeleteOptions{},
+			)
+
+			// As long as the customer can clean up resources on their own, we don't block deletion on failure to clean up federated credentials
+			if err != nil {
+				m.log.Errorf("failed to delete federated identity credentials for %s: %v", identity.ResourceID, err)
+			}
```

This is an error on the RP side and not related to the cluster MSI having been deleted by the cx, so I think we should error out here rather than logging the error and continuing.

Also, couldn't we skip the `GET` call right above this and simply rework this such that we continue if the `DELETE` results in a 404? We should be able to get the same result with one fewer API call.

---

updated